packages/teleport/src/Apps/AppList/AppList.tsx

```diff
@@ -1,106 +1,207 @@
-/**
- * Copyright 2020 Gravitational, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-import React from 'react';
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useState } from 'react';
 import styled from 'styled-components';
-import { Text, Flex } from 'design';
-import { MenuIcon, MenuItem } from 'shared/components/MenuAction';
+import { sortBy } from 'lodash';
+import { Flex, Text } from 'design';
+import {
+  pink,
+  teal,
+  cyan,
+  blue,
+  green,
+  orange,
+  brown,
+  red,
+  deepOrange,
+  blueGrey,
+} from 'design/theme/palette';
+import {
+  Column,
+  SortHeaderCell,
+  Cell,
+  renderLabelCell,
+  TextCell,
+  SortTypes,
+} from 'design/DataTable';
+import Table from 'design/DataTable/Paged';
+import isMatch from 'design/utils/match';
 import { App } from 'teleport/services/apps';
-import { NewTab } from 'design/Icon';
 
-export default function AppList({ apps = [] }: Props) {
-  const $apps = apps.map(app => <Item mb={4} mr={5} app={app} key={app.id} />);
-  return <Flex flexWrap="wrap">{$apps}</Flex>;
+function AppList(props: Props) {
+  const { apps = [], pageSize = 100, searchValue } = props;
+
+  const [sortDir, setSortDir] = useState<Record<string, string>>({
+    name: SortTypes.DESC,
+  });
+
+  function sortAndFilter(search) {
+    const filtered = apps.filter(obj =>
+      isMatch(obj, search, {
+        searchableProps: ['name', 'publicAddr', 'tags'],
+        cb: searchAndFilterCb,
+      })
+    );
+
+    const columnKey = Object.getOwnPropertyNames(sortDir)[0];
+    const sorted = sortBy(filtered, columnKey);
+    if (sortDir[columnKey] === SortTypes.ASC) {
+      return sorted.reverse();
+    }
+
+    return sorted;
+  }
+
+  function onSortChange(columnKey: string, sortDir: string) {
+    setSortDir({ [columnKey]: sortDir });
+  }
+
+  const data = sortAndFilter(searchValue);
```

Comment:
I would move this line `const data = sortAndFilter(searchValue);` below the func declarations (so functions are visually grouped)